snapcraft/plugins/jhbuild.py

```diff
@@ -0,0 +1,1035 @@
+                'type': 'string',
+                'default': 'current',
+            },
+            'container-name': {
+                'type': 'string',
+            },
+            'distribution': {
+                'type': 'string',
+                'default': platform.linux_distribution()[0].lower(),
+            },
+            'release': {
+                'type': 'string',
+                'default': platform.linux_distribution()[2].lower(),
+            },
+            'architecture': {
+                'type': 'string',
+                'default': {
+                    'amd64': 'amd64',
+                    'x86_64': 'amd64',
+                    'i386': 'i386',
+                    'i686': 'i386',
+                } [platform.machine()],
+            },
+            'container-packages': {
+                'type': 'array',
+                'items': {
+                    'type': 'string',
+                },
+                'uniqueItems': True,
+            },
+            'debmirror': {
+                'type': 'string',
+            },
+            'jhtarballs': {
+                'type': 'string',
+            },
+            'jhmirror': {
+                'type': 'string',
+            },
+            'jhcheckout': {
+                'type': 'string',
+            },
+            'ccache': {
+                'type': 'string',
+            },
+            'disable-pull': {
+                'type': 'boolean',
+                'default': False,
+            },
+            'disable-build': {
+                'type': 'boolean',
+                'default': False,
+            },
+            'disable-jhupdate': {
+                'type': 'boolean',
+                'default': False,
+            },
+            'disable-parallel': schema['properties']['disable-parallel'],
+        }
+
+        schema['required'] = [
+            'modules',
+            'snap-name',
+        ]
+
+        schema['pull-properties'] = [
+            'modules',
+            'moduleset',
+            'snap-name',
+            'snap-revision',
+            'container-name',
+            'distribution',
+            'release',
+            'architecture',
+            'container-packages',
+            'debmirror',
+            'jhtarballs',
+            'jhmirror',
+            'jhcheckout',
+            'ccache',
+            'disable-pull',
+            'disable-jhupdate',
+        ]
+
+        schema['build-properties'] = [
+            'disable-build',
+            'disable-parallel',
+        ]
+
+        return schema
+
+    def __init__(self, name, options, project=None):
+        super().__init__(name, options, project)
+
+        self.stage_packages += STAGE_PACKAGES
+
+    @property
+    def modules(self):
+        return [module for module in self.options.modules if not module.startswith('-')]
+
+    @property
+    def skipped_modules(self):
+        return [module[1:] for module in self.options.modules if module.startswith('-')]
```

Comment:
It's just to remove the initial '-' that's present for the skipped modules, like '-Webkit' -> 'WebKit'.